fennel/datasets/test_invalid_dataset.py

```diff
@@ -220,6 +221,77 @@ class RatingActivity:
     t: datetime
 
 
+def test_incorrect_assign_expr_type():
+    with pytest.raises(TypeError) as e:
+
+        @meta(owner="test@test.com")
+        @dataset
+        class RatingActivityTransformed:
+            userid: int
+            rating_sq: float
+            movie_suffixed: str
+            t: datetime
+
+            @pipeline
+            @inputs(RatingActivity)
+            def transform(cls, rating: Dataset):
+                return rating.assign(
+                    rating_sq=(F("rating") * F("rating")).astype(str),
+                    movie_suffixed=F("movie").str.concat("_suffix").astype(int),
+                ).drop("rating","movie")
+
+    assert (
+        str(e.value)
+        == """invalid assign - '[Pipeline:transform]->assign node' type errors: [TypeError("expression `(Field[`rating`] * Field[`rating`])` has type `<class 'float'>` does not match the type of column `rating_sq`, <class 'str'>"), TypeError('expression `Field[`movie`] + "_suffix"` has type `<class \\\'str\\\'>` does not match the type of column `movie_suffixed`, <class \\\'int\\\'>')]"""
+    )
```

Comment:
I think we should put it some work to significantly polish this error because it's going to be a common occurrence. Here is my attempt at what it should be:

`type error:  '{kwarg}' in assign operator of '{DS}.{pipeline}' is of type float, can not be promoted to str. Full expression: {expression}`

Reply:
Here i am combining all errors that belong to the same pipeline so will make it - 

type error(s) in assign operator of '{DS}.{pipeline}': 
1. '{kwarg}'  is of type float, can not be promoted to str. Full expression: {expression}
2. '{kwarg2}'  is of type int, can not be promoted to str. Full expression: {expression}